src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Everything/PatientEverythingService.cs

```diff
@@ -177,6 +238,155 @@ public class PatientEverythingService : IPatientEverythingService
             return new SearchResult(searchResultEntries, searchResult.ContinuationToken, searchResult.SortOrder, searchResult.UnsupportedSearchParameters);
         }
 
+        private void CheckForReplacedByLinks(string parentPatientId, SearchResultEntry parentPatientResource)
```

Comment:
Should these new methods also have EnsureArgs?

Reply:
Do we need EnsureArgs for private methods? Happy to add them in, but I noticed we don't have them for other private methods in the solution.

Reply:
I think it depends on input, if the value has been previously checked by another method then its fine